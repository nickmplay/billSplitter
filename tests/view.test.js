const { ViewBill } = require('../src/view');

const fakeID = 1462361249717;
const fakePerson1 = {id:fakeID, type:"more", amount:10};
const fakePerson2 = {id:fakeID + 1, type:"split", amount:0};
const share1 = {id: fakePerson1.id, type: "split", share: 40};
const share2 = {id: fakePerson2.id, type: "split", share: 40};

//practise using toMatchSnapshot with DOM manipulations
test('instantiate a ViewBill class', () => {
  const newVB = new ViewBill();
  expect( newVB.test() ).toMatchObject({id:"test"});
  expect( newVB.test() ).toMatchSnapshot();
}); 

test('test a ViewBill DOM return', () => {
  const newVB = new ViewBill();
  expect( newVB.domTest() ).toMatchSnapshot();
}); 

test('test a ViewBill DOM manipulation', () => {
  const newVB = new ViewBill();
  const div = document.createElement("div");
  div.setAttribute("class", "test-div");
  expect( div ).toMatchSnapshot();
  expect( newVB.domChangeTest(div) ).toMatchSnapshot();
}); 

//test person template injection
test('test person template injection', () => {
  const newVB = new ViewBill();
  const FP1Div = newVB.createPersonLi(fakePerson1);
  expect( FP1Div ).toMatchSnapshot();
}); 

//update person DOM
test('test person DOM update', () => {
  const newVB = new ViewBill();
  const fakePerson1a = {id:fakePerson1.id, type:"split", amount:12 };
  const FP1Div = newVB.createPersonLi(fakePerson1);
  expect( FP1Div ).toMatchSnapshot();
  newVB.updatePerson(FP1Div, fakePerson1a);
  expect( FP1Div ).toMatchSnapshot();
}); 

test('test person DOM does not update', () => {
  const newVB = new ViewBill();
  const fakePerson1a = {id:123, type:"split", amount:12 };
  const FP1Div = newVB.createPersonLi(fakePerson1);
  expect( FP1Div ).toMatchSnapshot();
  const flag = newVB.updatePerson(FP1Div, fakePerson1a);
  expect( FP1Div ).toMatchSnapshot();
  expect(flag).toBe(false);
}); 