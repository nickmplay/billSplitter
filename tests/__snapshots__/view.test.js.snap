// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instantiate a ViewBill class 1`] = `
Object {
  "id": "test",
}
`;

exports[`test a ViewBill DOM manipulation 1`] = `
<div
  class="test-div"
/>
`;

exports[`test a ViewBill DOM manipulation 2`] = `
<div
  class="new-class"
>
  <span>
    ViewBill entry
  </span>
</div>
`;

exports[`test a ViewBill DOM return 1`] = `
<div>
  <h1>
    Hello world
  </h1>
</div>
`;

exports[`test add person button 1`] = `
<button
  class="addPersonBtn"
>
  Add Person
</button>
`;

exports[`test default input amount box 1`] = `
<div
  class="totalBill"
>
  <label
    for="amountInp"
  >
    Enter total bill to split:
  </label>
  <input
    class="input-total"
    name="amountInp"
    placeholder="Enter amount"
    type="number"
    value="0"
  />
</div>
`;

exports[`test input amount box 1`] = `
<div
  class="totalBill"
>
  <label
    for="amountInp"
  >
    Enter total bill to split:
  </label>
  <input
    class="input-total"
    name="amountInp"
    placeholder="Enter amount"
    type="number"
    value="100"
  />
</div>
`;

exports[`test person DOM does not update 1`] = `
<li
  p-id="1462361249717"
>
  To pay 
  <input
    <=""
    class="p-amount"
    input=""
    placeholder="10"
    type="number"
  />
  <select>
    <option
      value="split"
    >
      split
    </option>
    <option
      selected=""
      value="more"
    >
      more
    </option>
    <option
      value="less"
    >
      less
    </option>
    <option
      value="fixed"
    >
      fixed
    </option>
  </select>
  <span
    p-id="1462361249717"
  >
    Share: -
  </span>
  <button>
    X
  </button>
</li>
`;

exports[`test person DOM does not update 2`] = `
<li
  p-id="1462361249717"
>
  To pay 
  <input
    <=""
    class="p-amount"
    input=""
    placeholder="10"
    type="number"
  />
  <select>
    <option
      value="split"
    >
      split
    </option>
    <option
      selected=""
      value="more"
    >
      more
    </option>
    <option
      value="less"
    >
      less
    </option>
    <option
      value="fixed"
    >
      fixed
    </option>
  </select>
  <span
    p-id="1462361249717"
  >
    Share: -
  </span>
  <button>
    X
  </button>
</li>
`;

exports[`test person DOM update 1`] = `
<li
  p-id="1462361249717"
>
  To pay 
  <input
    <=""
    class="p-amount"
    input=""
    placeholder="10"
    type="number"
  />
  <select>
    <option
      value="split"
    >
      split
    </option>
    <option
      selected=""
      value="more"
    >
      more
    </option>
    <option
      value="less"
    >
      less
    </option>
    <option
      value="fixed"
    >
      fixed
    </option>
  </select>
  <span
    p-id="1462361249717"
  >
    Share: -
  </span>
  <button>
    X
  </button>
</li>
`;

exports[`test person DOM update 2`] = `
<li
  p-id="1462361249717"
>
  To pay 
  <input
    <=""
    class="p-amount"
    input=""
    placeholder="12"
    type="number"
  />
  <select>
    <option
      selected=""
      value="split"
    >
      split
    </option>
    <option
      value="more"
    >
      more
    </option>
    <option
      value="less"
    >
      less
    </option>
    <option
      value="fixed"
    >
      fixed
    </option>
  </select>
  <span
    p-id="1462361249717"
  >
    Share: -
  </span>
  <button>
    X
  </button>
</li>
`;

exports[`test person template injection 1`] = `
<li
  p-id="1462361249717"
>
  To pay 
  <input
    <=""
    class="p-amount"
    input=""
    placeholder="10"
    type="number"
  />
  <select>
    <option
      value="split"
    >
      split
    </option>
    <option
      selected=""
      value="more"
    >
      more
    </option>
    <option
      value="less"
    >
      less
    </option>
    <option
      value="fixed"
    >
      fixed
    </option>
  </select>
  <span
    p-id="1462361249717"
  >
    Share: -
  </span>
  <button>
    X
  </button>
</li>
`;
